### ISSUE TITLE
MediaElement and MediaElementAudioSourceNode implementation

### ISSUE THREAD

**Issue by orottier** [orottier/web-audio-api-rs#120]
Lengthy discussion at #106. Two conflicting goals made us decide to postpone this issue and leave the functionality out of the v1 milestone.

The functionality was removed in commit 9e2264e
Commit be05c3e216dc6 contains the switch between taking ownership or just borrowing the media element inside the audio node

**Comment by orottier** [comment]
I think there is a decent solution that can satisfy both requirement.

The MediaElement will have a constructor alike:
```
impl MediaElement {
    /// Create a new MediaElement by buffering a MediaStream
    pub fn new<S: MediaStream, C: BaseAudioContext>(input: S, context: C, bs: BufferStrategy) -> Self;
```

By passing the audio context in the MediaElement constructor we can use the realtime audio callback to ensure progression of time of the media element. This callback not render anything but will make the current buffer of the stream available for reading.

Reading is done by the actual MediaElementAudioSourceNode that reference this element. There can be any number of them as they will just take the buffer that was made available and pass them into the audio graph.

BufferStrategy will be something clever like:
```
BufferStrategy {
    look_ahead: Duration
    look_back: Duration,
    max_in_memory_bytes: Option<usize>
    cache: Box<dyn Read + Write + Seek>
}
```

**Comment by orottier** [comment]
https://github.com/RustyDAW/creek may be interesting for this

**Comment by orottier** [comment]
Fixed with #168. 
Some features from the `MediaElement` are still missing, I will open a separate ticket for it
